src/lib/serviceRoutines/postUpdateContext.cpp

```diff
@@ -52,6 +53,321 @@ static char* xmlPayloadClean(const char*  payload, const char* payloadWord)
 
 /* ****************************************************************************
 *
+* forwardsPending - 
+*/
+static bool forwardsPending(UpdateContextResponse* upcrsP)
+{
+  for (unsigned int cerIx = 0; cerIx < upcrsP->contextElementResponseVector.size(); ++cerIx)
+  {
+    ContextElementResponse* cerP  = upcrsP->contextElementResponseVector[cerIx];
+
+    for (unsigned int aIx = 0 ; aIx < cerP->contextElement.contextAttributeVector.size(); ++aIx)
+    {
+      ContextAttribute* aP  = cerP->contextElement.contextAttributeVector[aIx];
+      
+      if (aP->providingApplication != "")
+      {
+        return true;
+      }      
+    }
+  }
+
+  return false;
+}
+
+
+
+/* ****************************************************************************
+*
+* updateForward - 
+*
+* An entity/attribute has been found on some context provider.
+* We need to forward the update request to the context provider, indicated in upcrsP->contextProvider
+*
+* 1. Parse the providing application to extract IP, port and URI-path
+* 2. Render an XML-string of the request we want to forward
+* 3. Send the request to the providing application (and await the response)
+* 4. Parse the XML response and fill in a binary UpdateContextResponse
+* 5. Render UpdateContextResponse from the providing application
+* 6. 'Fixing' StatusCode
+* 7. Freeing memory
+*/
+static void updateForward(ConnectionInfo* ciP, UpdateContextRequest* upcrP, UpdateContextResponse* upcrsP)
+{
+  std::string     ip;
+  std::string     protocol;
+  int             port;
+  std::string     prefix;
+  std::string     answer;
+
+
+  //
+  // 1. Parse the providing application to extract IP, port and URI-path
+  //
+  if (parseUrl(upcrP->contextProvider, ip, port, prefix, protocol) == false)
+  {
+    LM_W(("Bad Input (invalid providing application '%s')", upcrP->contextProvider.c_str()));
+    //  Somehow, if we accepted this providing application, it is the brokers fault ...
+    //  SccBadRequest should have been returned before, when it was registered!
+
+    upcrsP->errorCode.fill(SccContextElementNotFound, "");
+    return;
+  }
+
+
+  LM_M(("KZ: -----------------------------------------------"));
+  LM_M(("KZ: Forwarding to %s:%d", ip.c_str(), port));
+  upcrP->present("KZ1: ");
+
+
+  //
+  // 2. Render an XML-string of the request we want to forward
+  //
+  Format outFormat = ciP->outFormat;
+  ciP->outFormat = XML;
+  std::string  payload = upcrP->render(ciP, UpdateContext, "");
+  char*        cleanPayload;
+  ciP->outFormat = outFormat;
+
+  LM_M(("KZ: payload: %s", payload.c_str()));
+
+  if ((cleanPayload = xmlPayloadClean(payload.c_str(), "<updateContextRequest>")) == NULL)
```

Comment:
In the future we will have also JSON. May this could be generalized in the context of this PR? Or it is not possible and this would be modified in the future?